### ISSUE TITLE
NXdials isn't a NeXus base class

### ISSUE THREAD

**Issue by phyy-nx** [dials/dials#1124]
Data output by dials.export format=nxs (NXreflections format) includes a group named `dials`, with attribute `@NX_class=NXdials`, but NXdials doesn't exist in NeXus.

See:
https://github.com/dials/dials/blob/master/util/nexus/nx_mx.py#L735-L753

This came up during validation of the NXreflections file I've been working on here:
https://github.com/nexusformat/exampledata/tree/master/DLS/reflections/hdf5

See comment here:
https://github.com/nexusformat/exampledata/pull/19#issuecomment-580514755

Essentially, using NXdials is forbidden as the `NX` prefix is reserved.

Two possible resolutions:
- Easy: remove the NX_class attribute from the dials group. This will ensure the group is ignored by NeXus parsing software
- Better: figure out how the functionality in this group can either be found in NXmx/NXreflections already, or if there is something here that is missing in NeXus, propose additions.

Screenie for reference:
![image](https://user-images.githubusercontent.com/13471434/73500699-3f93aa80-4378-11ea-9acf-049b42e4c625.png)

**Comment by dagewa** [comment]
I much prefer the "better" option

**Comment by graeme-winter** [comment]
Only snag with the "better" option is it will probably take a year or two

Perhaps we do both? 

Remove NXclass in first instance, then try to migrate across where possible to NXproper and when we identify missing things, get Aaron to raise this.

It's something of a theoretical discussion at this time since nothing is using this output. 

I would also keep this distinct from dials use of HDF5 as a working format - this is an export / archive NeXus format we are discussing here ;-)

**Comment by graeme-winter** [comment]
@phyy-nx probably would not hurt to have a natter about this in a face-to-face like manner sometime - what are the ambitions, constraints etc. - in the happy land of CCP4 they have been looking for a "future reflection file format" for a little while now, which should perhaps be considered as part of the conversation?

**Comment by dagewa** [comment]
Gradual transition makes sense, but why should the archive format be distinct from the working format? I can understand that the working format might be a superset of the archive format, but where the same information is stored it should look the same, right?

**Comment by graeme-winter** [comment]
@dagewa - a reasonable perspective however means that for every minor change to our working intermediate files we need to get agreement from the NeXus committee (which has no representation from DIALS east) _or_ we have "parameters agreed by NeXus" in some data sets and "parameters not agreed by NeXus" in different data sets and need to match them up on reading the data. 

I can easily imagine us using precisely 

```
background.mean
background.sum.value
background.sum.variance
bbox
d
entering
flags
id
imageset_id
intensity.prf.value
intensity.prf.variance
intensity.sum.value
intensity.sum.variance
lp
miller_index
n_signal
num_pixels.background
num_pixels.background_used
num_pixels.foreground
num_pixels.valid
panel
partial_id
partiality
profile.correlation
qe
rlp
s1
shoebox
xyzcal.mm
xyzcal.px
xyzobs.mm.value
xyzobs.mm.variance
xyzobs.px.value
xyzobs.px.variance
zeta
```

as HDF5 dataset names - none of these are valid NeXus. If we chose to use HDF5 as a working format and have some fraction of these which are allowed in NXreflections and the rest in some other format I can see only confusion following.

**Comment by dagewa** [comment]
Quite a number of entries in your list exist (in some form) here: https://manual.nexusformat.org/classes/base_classes/NXreflections.html. Agree that adding the others to the standard makes sense. In the meantime I maintain that having a working format that is a superset of the export format is a practical approach while we wait for the committee to "catch up". It is less confusing than maintaining two distinct HDF5 reflection formats.

**Comment by phyy-nx** [comment]
Right, in fact IIRC, this concordance of DIALS column names with NeXus definitions in NXreflections was worked out in a NIAC meeting.  I took the proposal there and we hashed out a few revisions to the terminology and got it ratified.  It's not that hard to do it again if there is stuff missing.

It's cool to hear about CCP4 interest in an alternate format for reflections.

One of the design goals in NeXus is to allow users to put anything they want in the hdf5 file that is workflow or facility specific.  Then, as things become commonly needed across programs, to bring new things into the standard when they are settled.  I think that's what is being said here too.  So for now, I'll prep a version of dials.export which doesn't set the class attribute as NXdials, and then we can think about the longer approach of looking at what's in NXdials and moving it into NeXus.

**Comment by phyy-nx** [comment]
Ok, I've removed the NXdials tag for now. I'd like to leave this issue open for now though as we need to work out the 'better' option listed in the opening post.
